modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java

```diff
@@ -0,0 +1,384 @@
+            cleanup.add(raftMgr::stop);
+
+            cleanup.add(this::stopService);
+        }
+
+        void startService() {
+            // Refresh listener, to forget the old "last value".
+            raftGroupListener = new TestRaftGroupListener();
+
+            try {
+                serviceFuture = raftMgr.startRaftGroupNode(
+                        new RaftNodeId(GROUP_ID, new Peer(nodeName)),
+                        raftGroupConfiguration,
+                        raftGroupListener,
+                        RaftGroupEventsListener.noopLsnr,
+                        RaftGroupOptions.defaults().setLogStorageFactory(logStorageFactory)
+                );
+            } catch (NodeStoppingException e) {
+                fail(e.getMessage());
+            }
+        }
+
+        RaftGroupService getService() {
+            assertNotNull(serviceFuture);
+            assertThat(serviceFuture, willCompleteSuccessfully());
+
+            return serviceFuture.join();
+        }
+
+        void stopService() {
+            if (serviceFuture != null) {
+                serviceFuture = null;
+
+                try {
+                    raftMgr.stopRaftNode(new RaftNodeId(GROUP_ID, new Peer(nodeName)));
+                } catch (NodeStoppingException e) {
+                    fail(e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Tests a situation when one of the nodes in RAFT group lost its last update upon restart.
+     */
+    @Test
+    void testRestartSingleNode() throws Exception {
+        SimpleIgniteNode node0 = nodes.get(0);
+        SimpleIgniteNode node1 = nodes.get(1);
+        SimpleIgniteNode node2 = nodes.get(2);
+
+        assertThat(node1.getService().run(testWriteCommand("1")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "1");
+
+        node1.stopService();
+        node1.logStorage.truncateSuffix(1);
+        node1.startService();
+
+        waitForValueReplication(nodes, "1");
+
+        assertThat(node1.getService().run(testWriteCommand("2")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "2");
+
+        assertEquals(node0.logStorage.getLastLogIndex(), node1.logStorage.getLastLogIndex());
+        assertEquals(node1.logStorage.getLastLogIndex(), node2.logStorage.getLastLogIndex());
+    }
+
+    /**
+     * Tests a situation when 2 nodes out of 3 are being restarted, one of which experienced data loss in RAFT log.
+     */
+    @Test
+    void testRestartTwoNodes() throws Exception {
+        SimpleIgniteNode node0 = nodes.get(0);
+        SimpleIgniteNode node1 = nodes.get(1);
+        SimpleIgniteNode node2 = nodes.get(2);
+
+        assertThat(node1.getService().run(testWriteCommand("1")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "1");
+
+        node1.stopService();
+        node2.stopService();
+
+        node1.logStorage.truncateSuffix(1);
+
+        node2.startService();
```

Comment:
Shouldn't you wait for the node to finish starting up?

Reply:
Same reason, following `waitForValueReplication(nodes, "1");` is enough